vowpalwabbit/parse_args.cc

```diff
@@ -1325,7 +1325,16 @@ void parse_reductions(options_i& options, vw& all)
   all.reduction_stack.push(explore_eval_setup);
   all.reduction_stack.push(ExpReplay::expreplay_setup<'c', COST_SENSITIVE::cs_label>);
   all.reduction_stack.push(Search::setup);
-  all.reduction_stack.push(audit_regressor_setup);
+  // this string can be a constant under audit_regressor.h
+  all.reduction_stack.push([](options_i& options, vw& all) { return audit_regressor_setup("audit_regressor", options, all); });
```

Comment:
this could be  stack of tuples<string, setup_function> and setup_base can keep track if a nullptr got returned or not (which mean this function did something or not)

Reply:
nit: I personally prefer structs with named member vars vs tuples, I find the `std::get<0>()` ugly

Reply:
I'll probably move away from tuples and into a dictionary. I agree they are ugly in cpp.